metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertCommentCodeAction.scala

```diff
@@ -0,0 +1,170 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.codeactions.CodeAction
+import scala.meta.internal.parsing.Trees
+import scala.meta.pc.CancelToken
+import scala.meta.tokens.Token
+import scala.meta.tokens.Tokens
+
+import org.eclipse.{lsp4j => l}
+import scala.meta.io.AbsolutePath
+
+class ConvertCommentCodeAction(buffers: Buffers) extends CodeAction {
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  override def contribute(params: l.CodeActionParams, token: CancelToken)(
+      implicit ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+
+    (for {
+      content <- buffers.get(path)
+      tokens <- tokenizeIfNotRangeSelection(range, content)
+      codeAction <- createActionIfPossible(
+        path,
+        tokens,
+        range,
+        isSingleLineComment(content),
+      )
+    } yield codeAction).toList
+  }
+
+  private def tokenizeIfNotRangeSelection(range: l.Range, content: String) = {
+    Option
+      .when(range.getStart == range.getEnd)(
+        Trees.defaultTokenizerDialect(content).tokenize
+      )
+      .flatMap(_.toOption)
+  }
+
+  private def isSingleLineComment(content: String)(
+      t: Token
+  ): Boolean = t match {
+    case tc: Token.Comment =>
+      val currentLine = content.split('\n')(t.pos.startLine)
```

Comment:
Maybe we could split the `content` into lines only once when creating the `isSingleLineComment`, instead of redoing this work for each token? :slightly_smiling_face:

Reply:
definitely :sweat_smile: